### ISSUE TITLE
TL;DR works, but not the "long" way

### ISSUE THREAD

**Issue by AntonGue** [mluis/qemu-esp32#8]
As the title states, I can use the container as long as I use the "TL;DR" instructions.
But as soon as I try to build it from the git clone folder and run the command

"./flash.sh program/build/emulation.bin" 

I get an error:
"bash: ./flash.sh: cannot execute: required file not found"

I've seen someone had a similar issue, but it wasn't quite as mine, so I hope you could help me.
My idea was, if I'd be able to run the container that way, I may be able to flash my own firmware onto the emulation.?

**Comment by mluis** [comment]
Could you please paste the entire set of commands you're running? I don't think I'm able to replicate the issue.

**Comment by mluis** [comment]
Updated the README, please let me know how it works for you. Thanks.

**Comment by mluis** [comment]
Hi @AntonGue , I've just updated everything to the latest version and simplified stuff a bit. Let me know how it works for you. Thanks

**Comment by AntonGue** [comment]
Hi, thank you for the quick reply. I am not currently on my computer, so I am not able to try it now, but I will let you know as soon as I did.

**Comment by AntonGue** [comment]
Hi, I thank you for the quick reply.
I'm currently not at my computer, but I will let you know as soon as I tried it out. :) Have a great weekend

**Comment by AntonGue** [comment]
So here are the steps I took (Windows with Docker Desktop 24.0.6):

$ git clone https://github.com/mluis/qemu-esp32.git
$ cd qemu-esp32
$ docker build . -t qemu-esp32
$ docker run --name qemu-esp32 --rm -i -p 8080:80 -t qemu-esp32 bash
$ idf
$ idf.py build -C program
$ ./flash.sh program/build/emulation.bin

and I still get the error message:
bash: ./flash.sh: cannot execute: required file not found

What might be relevant: the docker image from mluis/qemu-esp32 is ~ 6.41 GB, and the one I get from the other way only ~6.31 GB, so I guess there might be something missing in the files?
I can't compare the "Dockerfile" from the docker-hub and github, so guess that may be a reason.

**Comment by AntonGue** [comment]
If it helps, I also get the following output when I run the tl;dr commands:

I (2244) main_task: Returned from app_main()
I (3224) esp_netif_handlers: eth ip: 10.0.2.15, mask: 255.255.255.0, gw: 10.0.2.2
W (20704) httpd_parse: parse_block: request URI/header too long
W (20704) httpd_txrx: httpd_resp_send_err: 431 Request Header Fields Too Large - Header fields are too long
W (20874) httpd_parse: parse_block: request URI/header too long
W (20874) httpd_txrx: httpd_resp_send_err: 431 Request Header Fields Too Large - Header fields are too long

As well as this on my browser, when I try to contact the localhost:
"Header fields are too long"

**Comment by AntonGue** [comment]
I figured out a little workaround:
I can use the -v option with an "mluis/qemu-esp32" image e.g.:

$ docker run -v C:/qemu-esp32:/flashing -it -p 8080:80 mluis/qemu-esp32

The command ./flash.sh the "flashing" folder (/flashing $ ./flash.sh program/build/emulation.bin) still won't work,
but it works, when I use it like this:

~# $ ./flash.sh /flashing/program/build/emulation.bin

**Comment by mluis** [comment]
ah got it, what happens if you prepend ./ like this: 
$ flash.sh ./program/build/emulation.bin ?

**Comment by AntonGue** [comment]
Thank you for your help. I found the reason and could solve the problem.
For anyone, who comes here later: https://unix.stackexchange.com/a/721845

tl;dr: the file was in the wrong format after cloning.
